include/nil/crypto3/zk/snark/arithmetization/plonk/detail/lookup_table_definition.hpp

```diff
@@ -154,12 +176,184 @@ namespace nil {
+                        std::size_t start_constant_column = 0;
+                        std::size_t prev_columns_number = 0;
+                        std::size_t full_selector_id = 0;
+
+                        for( const auto&table_name:ordered_table_names ){
+//                            std::cout << "Pack table " << std::endl;
+                            const auto &table = lookup_tables.at(table_name);
+
+                            if( table->get_rows_number() > max_usable_rows ){
+                                usable_rows_after = max_usable_rows;
+                                std::size_t options_number = table->get_rows_number() / max_usable_rows + 1;
+                                start_constant_column += prev_columns_number;
+                                std::size_t cur_constant_column = start_constant_column;
+                                prev_columns_number = options_number * table->get_columns_number();
+
+                                for(std::size_t i = start_constant_column; i < start_constant_column + prev_columns_number; i++){
+                                    constant_columns[i].resize(max_usable_rows);
+                                }
+
+                                if( full_selector_id == 0 ){
+                                    plonk_column<FieldType> selector_column(max_usable_rows, FieldType::value_type::one());
+                                    selector_column[0] = FieldType::value_type::zero();
+                                    assignment.fill_selector(cur_selector_id, selector_column);
+                                    full_selector_id = cur_selector_id;
+                                    cur_selector_id++;
+                                }
+
+                                if( table-> get_rows_number() % max_usable_rows == 0 ) options_number--;
+                                std::size_t cur = 0;
+                                for(std::size_t i = 0; i < options_number; i++){
+                                    for(std::size_t start_row = 1; start_row < max_usable_rows; start_row++, cur++){
+                                        for(std::size_t k = 0; k < table->get_columns_number(); k++){
+                                            if(cur < table->get_rows_number())
+                                                constant_columns[cur_constant_column + k][start_row] = table->get_table()[k][cur];
+                                            else
+                                                constant_columns[cur_constant_column + k][start_row] = table->get_table()[k][table->get_rows_number()-1];
+                                        }
+                                    }
+                                    cur_constant_column += table->get_columns_number();
+                                }
+                                for( const auto &[subtable_name, subtable]:table->subtables ){

```

Comment:
This part seems to be repeated below in the "not overflowing" case (aside from the assert), maybe it makes sense to move into a separate function?